src/safeposix/syscalls/fs_calls.rs

```diff
@@ -3271,64 +3271,168 @@ impl Cage {
+                        //filesystem table, we both disallow the creation of any files inside
+                        //that directory and prevent the process that has the directory
+                        //open from closing a nonexistent directory.
+                        //Setting the directory's linkcount as 2 works as a flag for
+                        //the `close_syscall()` to mark the directory that needs to be
+                        //removed from the filesystem  when its last open file descriptor 
+                        //is closed because it could not be removed at the time of calling
+                        //`rmdir_syscall()` because of some open file descriptor.
                         if remove_inode {
                             dir_obj.linkcount = 2;
-                        } // linkcount for an empty directory after rmdir must be 2
+                        }
+
+                        //The mutable reference to the inode has to be dropped because
+                        //remove_from_parent_dir() method will need to acquire an immutable 
+                        //reference to the parent directory's inode from the filesystem's 
+                        //inodetable. The inodetable represents a Rust DashMap that deadlocks
+                        //when trying to get a reference to its entry while holding any sort 
+                        //of reference into it.
                         drop(inodeobj);
 
+                        //`remove_from_parent_dir()` helper function returns 0 if an 
+                        //entry corresponding to the specified directory was 
+                        //successfully removed from the filename-inode dictionary 
+                        //of its parent.
+                        //If the parent directory does not grant write permission,
+                        //`EPERM` is returned.
+                        //As a sanity check, if the parent inode specifies a
+                        //non-directory type, the funciton panics
                         let removal_result =
                             Self::remove_from_parent_dir(parent_inodenum, &truepath);
                         if removal_result != 0 {
                             return removal_result;
                         }
 
-                        // remove entry of corresponding inodenum from inodetable
+                        //Remove entry of corresponding inodenum from the filesystem
+                        //inodetable
                         if remove_inode {
                             FS_METADATA.inodetable.remove(&inodenum).unwrap();
                         }
-
+                        //Log is used to store all the changes made to the filesystem. After
+                        //the cage is closed, all the collected changes are serialized and
+                        //the state of the underlying filsystem is persisted. This allows us
```

Comment:
small typo in the spelling of `filesystem`.

Reply:
fixed